recaf-ui/src/main/java/me/coley/recaf/ui/pane/SearchPane.java

```diff
@@ -294,4 +371,81 @@ private void searchDeclaration(String owner, String name, String desc, TextMatch
 		Search search = new Search().declaration(owner, name, desc, mode);
 		runSearch(search);
 	}
+
+	private class TextChangeCooldown implements Runnable {
+		private final BatchObservableSet<Result> results;
+		private final TextField txtText;
+		private final EnumComboBox<TextMatchMode> comboMode;
+
+		public TextChangeCooldown(BatchObservableSet<Result> results, TextField txtText, EnumComboBox<TextMatchMode> comboMode) {
+			this.results = results;
+			this.txtText = txtText;
+			this.comboMode = comboMode;
+		}
+
+		@Override
+		public void run() {
+			if (future == null)
+				return;
+			if (lastTyped.until(Instant.now(), ChronoUnit.MILLIS) <= 700) {
+				ThreadUtil.runDelayed(400, this);
+				return;
+			}
+			results.clear();
+			String text = txtText.getText();
+			if (!text.isEmpty()) {
+				runSearch(
+					new Search().text(text, comboMode.getValue()), results,
+					() -> {
+						results.triggerBatchListeners(true);
+						future = null;
+					}
+				);
+			}
+		}
+	}
+
+	private static class ResultListCell extends ListCell<Result> {
+
+		private final TextField txtText;
+
+		public ResultListCell(TextField txtText) {
+			this.txtText = txtText;
+			addEventFilter(MouseEvent.MOUSE_PRESSED, Event::consume);
+		}
+
+		@Override
+		protected void updateItem(Result item, boolean empty) {
+			super.updateItem(item, empty);
+			//						updateSelected(false);
+			setGraphic(null);
+			setText(null);
+			if (empty || !(item instanceof TextResult))
+				return;
```

Comment:
Generally format of `updateItem` used in the project is:

```java
if (empty || item == null) {
	setGraphic(null);
	setText(null);
} else {
	// standard population logic
}
```
This approach uses one less call to `setX` when a cell is recycled to fill something else vs resetting before each usage as the current logic does here.